apps/zbugs/server/get-query.ts

```diff
@@ -1,23 +1,26 @@
-import {withValidation, type ReadonlyJSONValue} from '@rocicorp/zero';
-import {queries as sharedQueries} from '../shared/queries.ts';
+import type {
+  AnyNamedQueryFunction,
+  AnyQuery,
+  ReadonlyJSONValue,
+} from '@rocicorp/zero';
 import {QueryError, QueryErrorCode} from '../shared/error.ts';
-import type {AuthData} from '../shared/auth.ts';
+import {queries as sharedQueries} from '../shared/queries.ts';
 
 // It's important to map incoming queries by queryName, not the
 // field name in queries. The latter is just a local identifier.
 // queryName is more like an API name that should be stable between
 // clients and servers.
-const validated = Object.fromEntries(
-  Object.values(sharedQueries).map(q => [q.queryName, withValidation(q)]),
+const queriesByQueryName = Object.fromEntries(
+  Object.values(sharedQueries).map(q => [q.queryName, q]),
 );
 
 export function getQuery(
-  context: AuthData | undefined,
   name: string,
   args: readonly ReadonlyJSONValue[],
-) {
-  if (name in validated) {
-    return validated[name](context, ...args);
+): AnyQuery {
+  if (name in queriesByQueryName) {
+    const f = queriesByQueryName[name] as AnyNamedQueryFunction;
```

Comment:
The validation is why we didn't need a cast here.

Reply:
the validation happens inside f.